appium/tests/specs/mock/setup/CheckRefreshKeyFromEKM.spec.ts

```diff
@@ -0,0 +1,183 @@
+  await SplashScreen.checkLoginPage();
+}
+
+describe('SETUP: ', () => {
+
+  it('app auto updates keys from EKM during startup with a pass phrase prompt', async () => {
+
+    const wrongPassPhraseError = CommonData.refreshKeys.wrongPassPhrase;
+    const passPhrase = CommonData.account.passPhrase;
+
+    const mockApi = new MockApi();
+    mockApi.fesConfig = {
+      clientConfiguration: {
+        flags: ["NO_PRV_CREATE", "NO_PRV_BACKUP", "NO_ATTESTER_SUBMIT", "PRV_AUTOIMPORT_OR_AUTOGEN", "FORBID_STORING_PASS_PHRASE"],
+        key_manager_url: ekmMockServer,
+      }
+    };
+    mockApi.ekmConfig = {
+      returnKeys: [ekmPrivateKeySamples.key0.prv]
+    }
+
+    await mockApi.withMockedApis(async () => {
+      // stage 1 - setup
+      await SplashScreen.login();
+      await SetupKeyScreen.setPassPhrase();
+      await goToKeysScreen();
+      await KeysScreen.checkKeysScreen([ekmPrivateKeySamples.key0]);
+
+      // stage 2 - prompt appears / wrong pass phrase rejected / cancel
+      mockApi.ekmConfig = {
+        returnKeys: [ekmPrivateKeySamples.key0.prv, ekmPrivateKeySamples.key1.prv]
+      }
+      await AppiumHelper.restartApp(processArgs);
+      await RefreshKeyScreen.waitForScreen(true);
+      await RefreshKeyScreen.fillPassPhrase('wrong passphrase');
+      await RefreshKeyScreen.clickOkButton();
+      await BaseScreen.checkModalMessage(wrongPassPhraseError);
+      await RefreshKeyScreen.clickOkButton();
+      await RefreshKeyScreen.cancelRefresh();
+      await goToKeysScreen();
+      await KeysScreen.checkKeysScreen([ekmPrivateKeySamples.key0]);
+
+      // stage 3 - new key gets added
+      await AppiumHelper.restartApp(processArgs);
+      await RefreshKeyScreen.waitForScreen(true);
+      await RefreshKeyScreen.fillPassPhrase(passPhrase);
+      await RefreshKeyScreen.clickOkButton();
+      await BaseScreen.checkToastMessage(successMessage);
+      await goToKeysScreen();
+      await KeysScreen.checkKeysScreen([ekmPrivateKeySamples.key0, ekmPrivateKeySamples.key1]);
+
+      // stage 4 - modified key gets updated, removed key does not get removed
+      mockApi.ekmConfig = {
+        returnKeys: [ekmPrivateKeySamples.key0Updated.prv]
+      }
+      await AppiumHelper.restartApp(processArgs);
+      await RefreshKeyScreen.waitForScreen(true);
+      await RefreshKeyScreen.fillPassPhrase(passPhrase);
+      await RefreshKeyScreen.clickOkButton();
+      await BaseScreen.checkToastMessage(successMessage);
+      await goToKeysScreen();
+      await KeysScreen.checkKeysScreen([ekmPrivateKeySamples.key0Updated, ekmPrivateKeySamples.key1]);
+
+      // stage 5 - older version of key does not get updated
+      mockApi.ekmConfig = {
+        returnKeys: [ekmPrivateKeySamples.key0.prv]
+      }
+      await AppiumHelper.restartApp(processArgs);
+      await goToKeysScreen();
+      await KeysScreen.checkKeysScreen([ekmPrivateKeySamples.key0Updated, ekmPrivateKeySamples.key1]);
+      await logoutUserFromKeyScreen();
```

Comment:
`logoutUserFromKeyScreen` is probably needed because there are three tests in one spec. So the state persists, I suppose?

It would probably be appropriate to split these three into three separate files (as other tests are) and then you don't have to log out in between. More importantly, if one test fails, it won't fail all three of these.

Reply:
Ok, got it